src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,75 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String NAME_SUPPLY = "supply";
+    private static final String NAME_BUY = "buy";
+    private static final String NAME_RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        File file = new File(toFileName);
+        if (file.exists()) {
+            return;
+        }
+        for (String row : getReport(fromFileName)) {
+            try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName, true))) {
+                writer.write(row + System.lineSeparator());
+            } catch (IOException e) {
+                throw new RuntimeException("Can't write data to file", e);
+            }
+        }
```

Comment:
Let's move this logic to a separate private method. It is responsible for writing data